JAGPropertyConverterTests/ExampleTest.m

```diff
@@ -714,4 +731,43 @@ - (void)testFromJSONDictWithCustomName {
     STAssertEquals(address.country, @"USA", @"age should be 55");
 }
 
+- (void)testFromJSONDictWithKeypathCustomName {
+    converter.classesToConvert = [NSSet setWithArray:@[[LivingAddress class]]];
+
+    LivingAddress *address = [[LivingAddress alloc] init];   // custom address has implemented JAGPropertyMappingProtocol
+    address.street = @"Infinite Loop 1";
+    address.city = @"Cuppertino";
+    address.country = @"USA";
+
+    Tenant *tenant1 = [[Tenant alloc] init];
+    tenant1.firstName = @"John";
+    tenant1.addressInformation = address;
+
+    Address *permanentAddress = [[Address alloc] init];
+    permanentAddress.street = @"Nowhere Road 2";
+    permanentAddress.city = @"Anywhere City";
+    permanentAddress.country = @"USA";
+    tenant1.permanentAddress = permanentAddress;
+
+    address.tenant = tenant1;
+
+    NSDictionary *targetDict = @{ @"country" : @"USA", @"city" : @"Cuppertino", @"street" : @"Infinite Loop 1", @"tenantName" : @"John", @"tenantPermanentAddressStreet": @"Nowhere Road 2" };
+    NSDictionary *userJsonDict = [converter decomposeObject:address];
+    STAssertEqualObjects(userJsonDict, targetDict, @"Converter decomposes model objects to JSON-compliant dictionaries.");
+}
+
+- (void)testToJSONWIthKeypathCustomName {
+    // newCustomProperty --> street
+    NSDictionary *sourceDict = @{ @"country" : @"USA", @"city" : @"Cuppertino", @"street" : @"Infinite Loop 1", @"tenantName" : @"John", @"tenantPermanentAddressStreet": @"Nowhere Road 2" };
```

Comment:
and again, create another test method where you use keypath directly in the dictionary.